plugins/modules/nd_federation_member.py

```diff
@@ -0,0 +1,280 @@
+    elif state == "absent":
+        if nd.existing:
+            if not module.check_mode:
+                for member_host in federation_member_obj:
+                    if not member_host.get("spec").get("host") == local_cluster_name:
+                        cluster_member_path = "{0}/{1}".format(member_path, member_host.get("status").get("memberID"))
+                        nd.request(cluster_member_path, method="DELETE")
+
+                # Remove the federation if there are no more members.
+                if len(nd.query_obj(member_path, ignore_not_found_error=True).get("items")) == 1:
+                    federation_info = next(
+                        (federation_dict for federation_dict in federation_obj if federation_dict.get("spec").get("name") == local_cluster_name), None
+                    )
+                    if federation_info:
+                        nd.request("{0}/{1}".format(federation_path, federation_info.get("status").get("federationID")), method="DELETE")
+            nd.existing = {}
+    elif state == "present":
+        remove_member_list = []
+        add_member_list = []
+        payload_dict = {}
+        if len(federation_member_obj) <= 1:
+            # if there are no members or just a local member, add all members from users.
+            add_member_list = clusters
+        else:
+            # Remove exisiting members not specified by the users.
+            member_exists = False
+            for existing_member_hosts in federation_member_obj:
+                if not existing_member_hosts.get("spec").get("host") == local_cluster_name:
+                    for user_member_host in clusters:
+                        if existing_member_hosts.get("spec").get("host") == user_member_host.get("cluster_hostname"):
+                            member_exists = True
+                            break
+                    if not member_exists:
+                        remove_member_list.append(existing_member_hosts)
+
+            # Add members specified by the users.
+            for user_member_host in clusters:
+                found = False
+                for existing_member_hosts in federation_member_obj:
+                    if existing_member_hosts.get("spec").get("host") == user_member_host.get("cluster_hostname"):
+                        found = True
+                        break
+                if not found:
+                    add_member_list.append(user_member_host)
+
+        if add_member_list or remove_member_list:
+            if remove_member_list:
+                for member in remove_member_list:
+                    cluster_member_path = "{0}/{1}".format(member_path, member.get("status").get("memberID"))
+                    nd.request(cluster_member_path, method="DELETE")
+
+                    try:
+                        payload_dict["DELETE"].append(cluster_member_path)
+                    except:
+                        payload_dict["DELETE"] = [cluster_member_path]
```

Comment:
Why do the try/except here and not initialise the payload dict with "DELETE" key and empty list? Same applies for POST.